tokio-postgres/src/binary_copy.rs

```diff
@@ -153,7 +153,7 @@ impl Stream for BinaryCopyOutStream {
             Some(header) => header.has_oids,
             None => {
                 check_remaining(&chunk, HEADER_LEN)?;
-                if &chunk.bytes()[..MAGIC.len()] != MAGIC {
+                if &chunk.chunk()[..MAGIC.len()] != MAGIC {
```

Comment:
You should be able to just do `if !chunk.starts_with(&MAGIC)` here I think.

Reply:
`chunk` is a `Cursor<Bytes>` which unfortunately does not implement `AsRef<[u8]>`. Switched to `!chunk.chunk().starts_with(&MAGIC)`. 

The trait method `Buf::chunk` is not documented to return the whole slice, but the implementation of `Buf` for `Cursor<Bytes>` does make that guarantee. If relying on that is scary, we could instead do `chunk.read_bytes(MAGIC.len()) != MAGIC`, which is safe for any implementation of `Buf`.

Reply:
Oh - we should just remove the Cursor layer entirely since Bytes implements Buf directly now.

Reply:
I tried that, but we need access to the cursor position: https://github.com/sfackler/rust-postgres/blob/master/tokio-postgres/src/binary_copy.rs#L201

Could always do some pointer arithmetic to compute the position, but didn’t seem like an obvious win.

Reply:
Ah right - should be fine as-is then.